code/Tree.cs

```diff
@@ -37,15 +37,90 @@ public class Tree
         }
 
         /// <summary>
+        /// Gives the number of children of a vertice.
+        /// </summary>
+        /// <param name="verticeId"> The id of the vertice parent. </param>
+        /// <returns> Returns the number of children vertices (0 if there are none).
+        /// If the vertice parent does not exist, it returns -1 .</returns>
+        public int NbChildren(int verticeId)
+        {
+            int nbChildren = 0;
+            Dictionary<int, List<int>> temporary = children;
+
+            if (!temporary.ContainsKey(verticeId))
+                nbChildren = -1;
```

Comment:
NbChildren has to be >= 0
In the case of the vertex do not belong to the Tree, return 0 or an error.
I would prefer 0.